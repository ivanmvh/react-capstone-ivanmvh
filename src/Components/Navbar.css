.navigationLink {
  text-decoration: none;
  color: black;
  font-weight: bold;
  font-size: 1.1rem;
}

.navigationLink:hover {
  transform: scale(1.1);
  color: white;
}

nav {
  height: 50px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: rgb(203, 65, 117);
}

#logo {
  height: 40px;
  margin-top: 10px;
  padding-right: 24px;
}

#navigation {
  display: none;
}

.currentPage {
  color: white;
  text-decoration: underline;
}

@media (min-width: 768px) {
  nav {
    height: 70px;
  }

  #navigation {
    width: 45%;
    padding-right: 5%;
    display: flex;
    justify-content: space-between;
  }

  #logo {
    height: 60px;
    width: 40%;
  }

  #burgerContainer {
    display: none;
  }
}

.buttonNav {
  border: none;
  background-color: transparent;
}

.header-items {
  color: white;
  font-size: '20px';
  margin-Top: '2px';
  font-weight: 800; 
}
