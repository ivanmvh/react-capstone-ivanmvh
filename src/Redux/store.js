import { combineReducers, configureStore } from '@reduxjs/toolkit';
import photosSlice from './PhotosSlice';

const photosReducers = combineReducers({
  Photos: photosSlice,
});

const store = configureStore({
  reducer: photosReducers,
});

export default store;
