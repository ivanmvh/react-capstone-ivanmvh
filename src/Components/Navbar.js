import './Navbar.css';
/* import { NavLink, useLocation, useNavigate } from 'react-router-dom'; */
import { NavLink, useLocation } from 'react-router-dom';
import { IoIosArrowBack, IoIosSettings } from 'react-icons/io';
import { BsFillMicFill } from 'react-icons/bs';

const Navbar = () => {
  /* const navigate = useNavigate(); */
  const location = useLocation();

  const currentYear = new Date().getFullYear();
  return (
    <nav>
      <div className="flex row">
        <NavLink
          className={({ isActive }) => (isActive ? 'activeLink' : 'homeLink')}
          to="/"
          type="button"
        >
          <IoIosArrowBack />
        </NavLink>
        <p id="year">{currentYear}</p>
      </div>
      <p className="header-items">{location.pathname === '/' ? 'most view categories' : 'photos views'}</p>
      <div>
        <button
          className="buttonNav"
          type="button"
        >
          <BsFillMicFill className="header-items" />
        </button>
        <button
          className="buttonNav"
          type="button"
        >
          <IoIosSettings className="header-items" />
        </button>
      </div>
    </nav>
  );
};

export default Navbar;
