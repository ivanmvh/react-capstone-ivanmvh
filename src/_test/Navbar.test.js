import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Components/Navbar';
import '@testing-library/jest-dom';
import store from '../Redux/store';

describe('Navbar', () => {
  const tree = test('renders Navbar component', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
