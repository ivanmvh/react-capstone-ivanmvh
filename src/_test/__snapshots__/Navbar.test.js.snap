// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar component 1`] = `undefined`;
